File: domain/src/main/java/com/thinkzi/weather/domain/utility/check/Checker.java
package com.thinkzi.weather.domain.utility.check;

/**
 * provide some data checks
 * */
public final class Checker {

    /**
    * check not null T object
    * */
    public static <T> T checkNotNull(T _objectReference) {

        if (_objectReference == null) {

            throw new NullPointerException();

        }

        return _objectReference;

    }

}

File: domain/src/main/java/com/thinkzi/weather/domain/utility/check/Checker.kt
package com.thinkzi.weather.domain.utility.check

/**
 * provide some data checks
 */
object Checker {

    /**
     * check not null T object
     */
    fun <T> checkNotNull(_objectReference: T?): T {

        if (_objectReference == null) {

            throw NullPointerException()

        }

        return _objectReference

    }

}
